src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -415,7 +413,17 @@ private static List<Object> expandDirectories(
         if (isDirectory(object)) {
           Artifact artifact = (Artifact) object;
           if (artifact.isTreeArtifact()) {
-            expandedValues.addAll(artifactExpander.expandTreeArtifact(artifact));
+            try {
+              expandedValues.addAll(artifactExpander.expandTreeArtifact(artifact));
+            } catch (MissingExpansionException e) {
+              throw new CommandLineExpansionException(
+                  String.format(
+                      "Failed to expand directory %s. Either add the directory as an input of the"
+                          + " action or set 'expand_directories = False' in the 'add_all' call"
```

Comment:
Do we know it's `add_all` or might it be another one like `add_joined`? Might be better to make the advice more general.

Reply:
You are right, it could also be `add_joined`, which I added to the error message. 

Both methods accept `expand_directories` and going by the linked issues, I see value in an error message that describes the (as far as I can tell only) sensible fixes as clearly as possible. But if this would risk missing a reasonable suggestion, making it more general instead would definitely be safer.